base-system/usrroot/usr/sbin/ipman

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+#	ipman.sh
+#	This script sets the IP address of the huronOS instance based on the
+#   configuration file (sync-server.conf) and if no interface sticks with
+#   the defined IP it leaves the available interfaces with dhcp.
+#
+#	Copyright (C) 2022, huronOS Project:
+#		<http://huronos.org>
+#
+#	Licensed under the GNU GPL Version 2
+#		<http://www.gnu.org/licenses/gpl-2.0.html>
+#
+#	Authors:
+#		Daniel Cerna <dcerna@huronos.org>
+
+. /usr/lib/hos/enviroment.sh
+
+# Retrieves the required fields
+IP_ADDRESS=$(grep "^IP=.*$" $CURRENT_SYNC_SERVER_CONFIG_FILE | cut -d= -f2)
+IP_GATEWAY=$(grep "^GATEWAY=.*$" $CURRENT_SYNC_SERVER_CONFIG_FILE | cut -d= -f2)
+IP_SERVER=$(grep "^DIRECTIVES_SERVER_IP=.*$" $CURRENT_SYNC_SERVER_CONFIG_FILE | cut -d= -f2)
+
+# Test if any of the required fields is empty
+if [ -z "$IP_ADDRESS" ]; then
+    echo "No IP defined, exiting"
+    exit 0
+fi
+if [ -z "$IP_GATEWAY" ]; then
+    echo "No gateway defined, exiting"
+    exit 0
+fi
+
+# Get a list of all available interfaces
+INTERFACES=$(connmanctl services | awk '{print $3}')
+
+# Loop through each interface and set the IP address
+for INTERFACE in $INTERFACES; do
+    # Sets the IP in the current interface
+    echo "Testing with interface $INTERFACE"
+    connmanctl config "$INTERFACE" --ipv4 manual "$IP_ADDRESS" 255.255.255.0 "$IP_GATEWAY"
+
+    # Wait for 3 seconds to allow the network to become fully operational
+    sleep 3
```

Comment:
Not sure if there's a way to actually check directly with connman.

Reply:
Connman seems to have a status command, but at lest in huronOS it stays at `State = Ready` if an IP is configured on the interface (even if is not a valid ip / there's no response from the gateway for any request), yet it does not transition to the `State = Online` even if the connection is successful and requests to the outside have a response.